### PR TITLE
Add option to specify enableLibraryEvolution

### PR DESCRIPTION
**Pull request by giginet**
This PR should be review after merging https://github.com/giginet/Scipio/pull/47

This PR added `--disable-library-evolution` flag to disable Library Evolution.
Please read README

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -21,7 +21,7 @@ let package = Package(
         .package(url: "https://github.com/apple/swift-log.git",
                  .upToNextMinor(from: "1.4.2")),
         .package(url: "https://github.com/apple/swift-argument-parser.git", 
-                 .upToNextMinor(from: "1.1.0")),
+                 from: "1.1.0"),
```
Comment (giginet) [inline Package.swift:24]: Become looser to fix resolver

File: Sources/scipio/Options.swift [hunk 2]
```diff
@@ -29,6 +29,11 @@ struct BuildOptionGroup: ParsableArguments {
           help: "Whether generated frameworks are Static Frameworks or not")
     var shouldBuildStaticFramework = false
 
+    @Flag(name: [.customLong("library-evolution")],
+          inversion: .prefixedEnableDisable,
```
Comment (giginet) [inline Sources/scipio/Options.swift:33]: Thanks to this option, swift-argument-parser automatically generates two flags `enable-library-evolution` and `disable-library-evolution`.

**Review by giuk-jung — APPROVED** [review]
LGTM

File: Sources/ScipioKit/Producer/PIF/PIFCompiler.swift [hunk 3]
```diff
@@ -115,7 +116,7 @@ struct PIFCompiler: Compiler {
             toolchain: toolchain,
             destinationTriple: toolchain.triple,
             flags: .init(),
-            enableParseableModuleInterfaces: true,
+            enableParseableModuleInterfaces: buildOptions.enableLibraryEvolution,
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/PIF/PIFCompiler.swift:119]: Why do we need to pass `buildOptions.enableLibraryEvolution` to `enableParseableModuleInterfaces`? I could not find any relation of them

File: Sources/ScipioKit/Runner.swift [hunk 4]
```diff
@@ -162,14 +166,16 @@ extension Runner {
                 isDebugSymbolsEmbedded: Bool? = nil,
                 frameworkType: FrameworkType? = nil,
                 extraFlags: ExtraFlags? = nil,
-                extraBuildParameters: [String: String]? = nil
+                extraBuildParameters: [String: String]? = nil,
+                enableLibraryEvolution: Bool? = false
```
Comment (freddi-kit) [inline Sources/ScipioKit/Runner.swift:170]: https://github.com/giginet/Scipio/blob/980d9cebc4276f3ea9bd6aaa7c6f89179b4397ac/Sources/ScipioKit/Runner.swift#L140

On the other hand, the similar default value is true, please unify or add comment about it to make clear

File: Sources/ScipioKit/Producer/PIF/PIFCompiler.swift (hunk 3 above)
Comment (giginet) [reply Sources/ScipioKit/Producer/PIF/PIFCompiler.swift:119]: Sorry, I explained the context here. https://github.com/giginet/Scipio/pull/47/files#r1106665406

`enableParseableModuleInterfaces` indicates whether enabling library-evolution or not internally.
https://github.com/apple/swift-package-manager/blob/2dc6407d38bc8d2a864baec035029a29fb36d146/Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift#L466

However, this value doesn't effect our product.

File: Sources/ScipioKit/Runner.swift (hunk 4 above)
Comment (giginet) [reply Sources/ScipioKit/Runner.swift:170]: Nice catch. it should be `nil`.

```suggestion
                enableLibraryEvolution: Bool? = nil
```

**Review by freddi-kit — APPROVED** [review]

**Pull request merged**